### PR TITLE
Improve security checks

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

File: src/key.rs [hunk 1]
```diff
@@ -127,6 +126,11 @@ impl Serialize for PrivateKey {
     }
 
     fn from_bytes(raw: &[u8]) -> Result<Self, Error> {
+        const FR_SIZE: usize = (Fr::NUM_BITS as usize + 8 - 1) / 8;
+        if raw.len() != FR_SIZE {
```
Comment (DrPeterVanNostrand) [inline src/key.rs:130]: Is it possible that `raw` is a stream of bytes containing more than one key (when `raw.len() % FR_SIZE == 0 && raw.len() > FR_SIZE`)? Is it possible that we call `PublicKey::from_btyes(raw)` on `raw` multiple times to read multiple sequential keys?

File: src/key.rs [hunk 2]
```diff
@@ -127,6 +126,11 @@ impl Serialize for PrivateKey {
     }
 
     fn from_bytes(raw: &[u8]) -> Result<Self, Error> {
+        const FR_SIZE: usize = (Fr::NUM_BITS as usize + 8 - 1) / 8;
```
Comment (DrPeterVanNostrand) [inline src/key.rs:129]: It's not clear to me what the `+ 8 - 1` is doing

Comment (dignifiedquire) [reply src/key.rs:129]: ceil(size/8) without using floats

File: src/key.rs (hunk 1 above)
Comment (dignifiedquire) [reply src/key.rs:130]: if we do so, I never intended the api to be used like that

**Pull request merged**